### DISCUSSION TITLE
How can I check if matching trigger is already expanded?

### DISCUSSION THREAD

**Discussion by boltlessengineer** [L3MON4D3/LuaSnip#1185] [Q&A] [answered]
I want to implement autopair functionality with luasnip. e.g. `()|` expands to `(|)` where `|` is cursor.
But when I expand the snippet and move to jump index 0 (`()|`), because there is valid trigger characters in front of cursor so I get macro expanded again instead of jumping

```
if some_func()| then
  // 2nd jump index of if … then … end node
end
```
As I’ve already expanded & closed the `()` snippet, I want to jump to next if statement snippet jump index.
I don’t want to re-expand the `()` snippet

To do that, I want to check if cursor is at 0th jump index of `()` snippet.
How can I check the cursor is at the end of specific expanded snippet?

**Comment by L3MON4D3** [comment, marked as answer]
Mhmm, we don't really have a reliable api for the general case (for any cursor-position, find whether it is immediately behind some active snippet), but for this specific case where you've just jumped to the $0, you can get the trigger of the snippet before the cursor, and whether you've just jumped to the $0
```lua
local current_node = require("luasnip").session.current_nodes[vim.api.nvim_get_current_buf()]
local current_snippet_trigger = current_node.parent.snippet.trigger
local current_is_0 = current_node.pos == 0
```
(maybe, also check whether the cursor is actually still on the $0, get the position via `current_node:get_buf_position()`, and check it against the position of the cursor (beware of 0/1-based row-column-indexing))

**Reply by boltlessengineer** [reply]
Thanks, this seems to be working!

```lua

condition = function (_line_to_cursor, matched_trigger)
    local current_node = ls.session.current_nodes[vim.api.nvim_get_current_buf()]
    if not current_node then
        return true
    end
    local is_same_trigger = current_node.parent.snippet.trigger == matched_trigger
    local current_is_0 = current_node.pos == 0
    if not (is_same_trigger and current_is_0) then
        return true
    end
    local node_pos = current_node:get_buf_position()
    local cur_pos = vim.api.nvim_win_get_cursor(0)
    local same_pos = node_pos[1] + 1 == cur_pos[1] and node_pos[2] == cur_pos[2]
    return not same_pos
end
```
